tests/test_toolchain.py

```diff
@@ -62,8 +70,11 @@ def test_create(self):
         ) as m_get_ndk_platform_dir, mock.patch(
             'pythonforandroid.build.get_cython_path'
         ) as m_get_cython_path, mock.patch(
-            'pythonforandroid.toolchain.build_dist_from_args'
-        ) as m_build_dist_from_args:
+            'pythonforandroid.toolchain.build_recipes'
+        ) as m_build_recipes, mock.patch(
+            'pythonforandroid.bootstraps.service_only.'
+            'ServiceOnlyBootstrap.run_distribute'
+        ) as m_run_distribute:
```

Comment:
This is just a generic gut feeling that might be wrong, but:

My intuitive reaction here is: this seems like this is mocked too much and will be a huge pain once we change some things. (I also find at some point it gets quite challenging to read)

Is there really no easier way to call this, maybe e.g. actually more on a real directory with test data and just check the output? I know this goes a bit back to me suggesting `subprocess` instead of mocking but here it seems to actually grow to such a size that I'm wondering if here it might be justified, it just looks a little like we're testing a too big component here in a mock way that is too large to be reasonably mock-tested in isolation

Reply:
Okay I thought about this more, and what I said is nonsense :joy: creating a distribution will compile all the recipes, right? So there is no quick way to test that in isolation and what you're doing makes perfect sense. I guess I just hoped there was an easier, less depending-exactly-on-so-many-specific-function-calls kind of way, but oh well :woman_shrugging: